controllers/topology/conditions_test.go

```diff
@@ -0,0 +1,446 @@
+			cluster:      &clusterv1.Cluster{},
+			s: &scope.Scope{
+				Blueprint: &scope.ClusterBlueprint{
+					Topology: &clusterv1.Topology{
+						Version: "v1.22.0",
+					},
+				},
+				Current: &scope.ClusterState{
+					Cluster: &clusterv1.Cluster{},
+					ControlPlane: &scope.ControlPlaneState{
+						Object: builder.ControlPlane("ns1", "controlplane1").
+							WithVersion("v1.22.0").
+							WithReplicas(3).
+							Build(),
+					},
+					MachineDeployments: scope.MachineDeploymentsStateMap{
+						"md0": &scope.MachineDeploymentState{
+							Object: builder.MachineDeployment("ns1", "md0-abc123").
+								WithReplicas(2).
+								WithStatus(clusterv1.MachineDeploymentStatus{
+									Replicas:          int32(2),
+									UpdatedReplicas:   int32(2),
+									ReadyReplicas:     int32(2),
+									AvailableReplicas: int32(2),
+								}).
+								Build(),
+						},
+					},
+				},
+				UpgradeTracker: func() *scope.UpgradeTracker {
+					ut := scope.NewUpgradeTracker()
+					ut.ControlPlane.PendingUpgrade = false
+					ut.ControlPlane.IsUpgrading = true
+					ut.MachineDeployments.MarkDeferredUpgrade("md0-abc123")
+					return ut
+				}(),
+			},
+			wantConditionStatus: corev1.ConditionFalse,
+			wantConditionReason: clusterv1.TopologyReconciledMachineDeploymentsUpgradePendingReason,
+		},
+		{
+			name:         "should set the condition to false if control plane picked the new version but machine deployments did not because control plane is scaling",
+			reconcileErr: nil,
+			cluster:      &clusterv1.Cluster{},
+			s: &scope.Scope{
+				Blueprint: &scope.ClusterBlueprint{
+					Topology: &clusterv1.Topology{
+						Version: "v1.22.0",
+					},
+				},
+				Current: &scope.ClusterState{
+					Cluster: &clusterv1.Cluster{},
+					ControlPlane: &scope.ControlPlaneState{
+						Object: builder.ControlPlane("ns1", "controlplane1").
+							WithVersion("v1.22.0").
+							WithReplicas(3).
+							Build(),
+					},
+					MachineDeployments: scope.MachineDeploymentsStateMap{
+						"md0": &scope.MachineDeploymentState{
+							Object: builder.MachineDeployment("ns1", "md0-abc123").
+								WithReplicas(2).
+								WithStatus(clusterv1.MachineDeploymentStatus{
+									Replicas:          int32(2),
+									UpdatedReplicas:   int32(2),
+									ReadyReplicas:     int32(2),
+									AvailableReplicas: int32(2),
+								}).
+								Build(),
+						},
+					},
+				},
+				UpgradeTracker: func() *scope.UpgradeTracker {
+					ut := scope.NewUpgradeTracker()
+					ut.ControlPlane.PendingUpgrade = false
+					ut.ControlPlane.IsScaling = true
+					ut.MachineDeployments.MarkDeferredUpgrade("md0-abc123")
+					return ut
+				}(),
+			},
+			wantConditionStatus: corev1.ConditionFalse,
+			wantConditionReason: clusterv1.TopologyReconciledMachineDeploymentsUpgradePendingReason,
+		},
+		{
+			name:         "should set the condition to true if control plane picked the new version and is upgrading but there are no machine deployments",
```

Comment:
```suggestion
			name:         "should set the condition to true if control plane picked the new version and is scaling but there are no machine deployments",
```
I think

Reply:
Here the message is correct but the test is wrong. Will fix it.